low-latency-networktables.adoc

```diff
@@ -0,0 +1,181 @@
+= Low Latency NetworkTables
+:toc: macro
+:sectanchors:
+:source-highlighter: pygments
+:pygments-style: colorful
+:networktables3: https://github.com/wpilibsuite/ntcore/blob/master/doc/networktables3.adoc
+
+toc::[]
+
+== Summary
+
+This change to NetworkTables provides a way to flag individual values
+as low latency and update them more frequently and synchronously
+across the network.  This is done by transmitting updates to these
+values via UDP (rather than TCP) and providing a way to signal
+flushing of these values to the network.
+
+== Motivation
+
+The current NetworkTables protocol is implemented using only TCP.  To
+limit network traffic in the case of network congestion or dropouts,
+value updates are rate limited in software to once every 100 ms.  For
+the vast majority of values (such as configuration parameters) this
+latency is not an issue.  However, it is often desirable to update a
+handful of values in a more real-time, low latency fashion, such as
+when doing image processing (where a new frame may be processed every
+30 ms).
+
+== Design
+
+=== Protocol Specification Changes
+
+This section describes the required changes to the
+{networktables3}[NetworkTables 3.0 Protocol].  There are no changes to
+the initial handshake (other than the version number) or other TCP
+operations.  The only operational change is to send Entry Update
+messages for Low-Latency flagged entries via UDP rather than TCP.
+
+==== Protocol Revision
+
+Protocol revision greater than or equal to 3.1 (0x0301) indicate
+support for low-latency messaging.
+
+==== Transport Layer
+
+The Server should listen on UDP port 1735 for incoming low-latency
+messages.  The Client should listen on a UDP port starting at port
+1735 (to allow for multiple Clients on a single machine, higher port
+numbers, e.g. 1736, 1737, etc. may need to be used).
+
+==== Client Hello
```

Comment:
Wouldn't the Client UDP receive from server port make more sense to come from the server in the Server Hello message? New clients are going to have no idea about other clients open. The server, on receiving a Client Hello, could enumerate its connected clients, and if it detects on from the same IP Address it can send a different Udp port back to the client for the client to listen on. 
cleaner to use.